client/ayon_resolve/plugins/publish/extract_editorial_package.py

```diff
@@ -0,0 +1,190 @@
+                f"Duration: {timeline_duration}, "
+                f"FPS: {timeline_fps}"
+            )
+
+            # export otio representation
+            self.export_otio_representation(
+                get_current_project(), timeline, otio_file_path
+            )
+
+        # Find Intermediate file representation file name
+        published_file_path = None
+        for repre in instance.data["representations"]:
+            if repre["name"] == "intermediate":
+                published_file_path = self._get_published_path(instance, repre)
+                break
+
+        if published_file_path is None:
+            raise ValueError("Intermediate representation not found")
+
+        # Finding clip references and replacing them with rootless paths
+        # of video files
+        otio_timeline = otio.adapters.read_from_file(otio_file_path.as_posix())
+        for track in otio_timeline.tracks:
+            current_end_frame = timeline_start_frame
+            for clip in track:
+                # skip transitions
+                if isinstance(clip, otio.schema.Transition):
+                    continue
+                # skip gaps
+                if isinstance(clip, otio.schema.Gap):
+                    # get duration of gap
+                    current_end_frame += clip.source_range.duration.value
+                    continue
+
+                if hasattr(clip.media_reference, "target_url"):
+                    path_to_media = Path(published_file_path)
+                    # remove root from path
+                    success, rootless_path = anatomy.find_root_template_from_path(  # noqa
+                        path_to_media.as_posix()
+                    )
+                    if success:
+                        media_source_path = rootless_path
+                    else:
+                        media_source_path = path_to_media.as_posix()
+
+                    new_media_reference = otio.schema.ExternalReference(
+                        target_url=media_source_path,
+                        available_range=otio.opentime.TimeRange(
+                            start_time=otio.opentime.RationalTime(
+                                value=timeline_start_frame, rate=timeline_fps
+                            ),
+                            duration=otio.opentime.RationalTime(
+                                value=timeline_duration, rate=timeline_fps
+                            ),
+                        ),
+                    )
+                    clip.media_reference = new_media_reference
+
+                    # replace clip source range
+                    new_source_range = otio.opentime.TimeRange(
+                        start_time=otio.opentime.RationalTime(
+                            value=current_end_frame, rate=timeline_fps
+                        ),
+                        duration=clip.source_range.duration,
+                    )
+                    clip.source_range = new_source_range
```

Comment:
I'm not sure I understand why the clip source range needs to be edited ?
Is it because swaping the media reference reset the source range (and if so isn't `new_source_range` always equal initial `clip.range_in_parent()` ?

Reply:
The issue is that we are relinking the original individual media sources and their source ranges onto one media source rendered from the timeline. There might be an easier solution you can suggest. This was the best I could find at the moment.

Reply:
I understand better yes. 
If the swapped media is the one rendered from the timeline, then media clip source range should always `clip.range_in_parent()`. This might save you some computation here.

Reply:
Thanks @robin-ynput, here is the commit with change 61012e27bcce5a91ea748d77856bb62c09b69637